cmd/clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -18,38 +18,16 @@ package clusterdeployer
 
 import (
 	"fmt"
-	"io/ioutil"
-	"os"
 	"strings"
-	"time"
 
-	"k8s.io/client-go/kubernetes"
 	"k8s.io/klog"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/clusterdeployer/bootstrap"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/clusterdeployer/clusterclient"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/clusterdeployer/provider"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/phases"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
-	"sigs.k8s.io/cluster-api/pkg/util"
 )
 
-// Deprecated interface for Provider specific logic. Please do not extend or add. This interface should be removed
-// once issues/158 and issues/160 below are fixed.
-type ProviderDeployer interface {
```

Comment:
Some providers test the interface in this package, the PR will need a release note to make sure folks update the interface package.